Gateway/Request/PaymentDataBuilder.php

```diff
@@ -55,20 +64,23 @@ public function build(array $buildSubject): array
     {
         /** @var PaymentDataObject $paymentDataObject */
         $paymentDataObject = SubjectReader::readPayment($buildSubject);
-
-        $order = $paymentDataObject->getOrder();
         $payment = $paymentDataObject->getPayment();
         $fullOrder = $payment->getOrder();
-
+        $quote = $this->checkoutSession->getQuote();
         $amountCurrency = $this->chargedCurrency->getOrderAmountCurrency($fullOrder);
         $currencyCode = $amountCurrency->getCurrencyCode();
         $amount = $amountCurrency->getAmount();
-        $reference = $order->getOrderIncrementId();
+        $reference = $fullOrder->getIncrementId();
+
+        $shopperConversionIdData = $quote->getPayment()->getAdditionalInformation('shopper_conversion_id');
+
+        $shopperConversionId = !empty($shopperConversionIdData) ? (string) json_decode($shopperConversionIdData, true) : '';
```

Comment:
Referring to my previous comment on the helper class; this block can be simplified if encoding the shopper conversion ID as JSON is redundant on the generator.